save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/LanguageSelector.kt

```diff
@@ -22,12 +24,16 @@ import web.cssom.Cursor
 
 private const val LANG_DROPDOWN_ID = "lang-dropdown"
 
+internal const val PREFERRED_LANGUAGE_COOKIE = "language"
+
 /**
  * A [FC] that is responsible for language selection
  */
 val languageSelector: VFC = FC {
     val (_, i18n) = useTranslation()
-    val (language, setSelectedLanguage) = useState(PlatformLanguages.defaultLanguage)
+    val languageFromCookie = PlatformLanguages.getByCodeOrDefault(cookie.get(PREFERRED_LANGUAGE_COOKIE))
```

Comment:
when you save cookies, you need to have a notification on frontend...

Reply:
Will do it in next PR if you don't mind